scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/NovaseqXProcessor.java

```diff
@@ -0,0 +1,241 @@
+  static final ObjectMapper MAPPER = new ObjectMapper();
+
+  BCLConvertAnalysis BCLConvertAnalyses = new BCLConvertAnalysis();
+
+  public NovaseqXProcessor(Builder builder, boolean checkOutput) {
+    super(builder, checkOutput);
+  }
+
+  @Override
+  public NotificationDto process(File runDirectory, TimeZone tz) throws IOException {
+    IlluminaDragenNotificationDto dto = new IlluminaDragenNotificationDto();
+    dto.clone((IlluminaNotificationDto) super.process(runDirectory, tz));
+    dto.setAnalysisStatus(AnalysisStatus.PENDING);
+    ObjectNode json = MAPPER.createObjectNode();
+    File analysisDir = new File(runDirectory, "Analysis");
+
+    // For n in Analysis/n/Data (accommodate reruns, ish. if more reruns appear, they won't be
+    // scanned. Someone will need to invalidate the run with the API.)
+    if (analysisDir.exists() && analysisDir.isDirectory()) {
+      // Null pointer should never actually happen because of above checks
+      for (File analysisAttempt : Objects.requireNonNull(analysisDir.listFiles())) {
+        if (analysisAttempt.isDirectory() && analysisAttempt.getName().matches(NUMERAL)) {
+          String attemptNum = analysisAttempt.getName();
+          ObjectNode jsonAttempt = MAPPER.createObjectNode();
+          Map<DRAGENWorkflow, Boolean> expectedWorkflows = new HashMap<>();
+
+          // Get info from Sample Sheet
+          // at root Analysis/#/Data/BCLConvert/SampleSheet.csv
+          File sampleSheet = new File(analysisAttempt, "Data/BCLConvert/SampleSheet.csv");
+          if (sampleSheet.exists()) {
+            List<String[]> lines =
+                Files.readAllLines(sampleSheet.toPath())
+                    .stream()
+                    .map(line -> line.split(","))
+                    .filter(line -> line.length != 0)
+                    .toList();
+
+            ObjectNode jsonSampleInfo = MAPPER.createObjectNode(),
+                sampleSheetBCLConvertSection = MAPPER.createObjectNode(),
+                sampleSheetBCLConvertData = MAPPER.createObjectNode(),
+                sampleSheetBCLConvertSettings = MAPPER.createObjectNode();
+            int s = 0;
+            String sectionName = "";
+            for (String[] line : lines) {
+              if (line[0].startsWith("[")) {
+                sectionName = line[0];
+                s = 0;
+                continue;
+              }
+              switch (sectionName) {
+                case "[BCLConvert_Data]":
+                  if (line[0].equals("Lane")) break; // Skip column label line
+                  s++; // Hopefully the lines stream in a consistent order and this will recreate
+                  // the S#
+                  ObjectNode nested = MAPPER.createObjectNode();
+                  nested.put("Lane", line[0]);
+                  nested.put("Sample_ID", line[1]);
+                  nested.put("Index", line[2]);
+                  nested.put("Index2", line[3]); // Some have a line[4] like "Y27;I10;I10;Y27"
```

Comment:
For exporting the sample sheet (which this code is doing), I think it will be more flexible going forward to keep the same formatting as the sample sheet rather than needing to convert between two sets of keys

Reply:
The strings are hard-coded here, so I don't think it really makes a difference in that regard. It's better that Run Scanner output is consistent